### ISSUE TITLE
[BUG] Error when normalizing numeric column with negatives - ValueError: Number of Input Columns must be same replacement Dataframe

### ISSUE THREAD

**Issue by gabrielspmoreira** [NVIDIA/NVTabular#432]
**Describe the bug**
This error occurs when I include a specific float32 column (NetworkTestLatency) in the workflow and apply the `nvt.ops.Normalize()`.
The column has negatives (-1.0) and it seems the error is related to computing `sqrt()` for negatives, which is invalid.

```
/opt/conda/envs/rapids/lib/python3.7/site-packages/pandas/core/series.py:726: RuntimeWarning: invalid value encountered in sqrt
  result = getattr(ufunc, method)(*inputs, **kwargs)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<timed eval> in <module>

/nvtabular0.3/NVTabular/nvtabular/workflow.py in apply(self, dataset, apply_offline, record_stats, shuffle, output_path, output_format, out_files_per_proc, num_io_threads, dtypes)
    869                 out_files_per_proc=out_files_per_proc,
    870                 num_io_threads=num_io_threads,
--> 871                 dtypes=dtypes,
    872             )
    873         else:

/nvtabular0.3/NVTabular/nvtabular/workflow.py in build_and_process_graph(self, dataset, end_phase, output_path, record_stats, shuffle, output_format, out_files_per_proc, apply_ops, num_io_threads, dtypes)
    968             self._base_phase = 0  # Set _base_phase
    969             for idx, _ in enumerate(self.phases[:end]):
--> 970                 self.exec_phase(idx, record_stats=record_stats, update_ddf=(idx == (end - 1)))
    971             self._base_phase = 0  # Re-Set _base_phase
    972 

/nvtabular0.3/NVTabular/nvtabular/workflow.py in exec_phase(self, phase_index, record_stats, update_ddf)
    755         _ddf = self.get_ddf()
    756         if transforms:
--> 757             _ddf = self._aggregated_dask_transform(_ddf, transforms)
    758 
    759         stats = []

/nvtabular0.3/NVTabular/nvtabular/workflow.py in _aggregated_dask_transform(self, ddf, transforms)
    724         for transform in transforms:
    725             columns_ctx, cols_grp, target_cols, logic, stats_context = transform
--> 726             meta = logic(meta, columns_ctx, cols_grp, target_cols, stats_context)
    727         return ddf.map_partitions(self.__class__._aggregated_op, transforms, meta=meta)
    728 

/nvtabular0.3/NVTabular/nvtabular/ops/transform_operator.py in apply_op(self, gdf, columns_ctx, input_cols, target_cols, stats_context)
     89         new_gdf = self.op_logic(gdf, target_columns, stats_context=stats_context)
     90         self.update_columns_ctx(columns_ctx, input_cols, new_gdf.columns, target_columns)
---> 91         return self.assemble_new_df(gdf, new_gdf, target_columns)
     92 
     93     def assemble_new_df(self, origin_gdf, new_gdf, target_columns):

/nvtabular0.3/NVTabular/nvtabular/ops/transform_operator.py in assemble_new_df(self, origin_gdf, new_gdf, target_columns)
     96                 return new_gdf
     97             else:
---> 98                 origin_gdf[target_columns] = new_gdf
     99                 return origin_gdf
    100         return cudf.concat([origin_gdf, new_gdf], axis=1)

/opt/conda/envs/rapids/lib/python3.7/contextlib.py in inner(*args, **kwds)
     72         def inner(*args, **kwds):
     73             with self._recreate_cm():
---> 74                 return func(*args, **kwds)
     75         return inner
     76 

/opt/conda/envs/rapids/lib/python3.7/site-packages/cudf/core/dataframe.py in __setitem__(self, arg, value)
    777                         replace_df=value,
    778                         input_cols=arg,
--> 779                         mask=None,
    780                     )
    781                 else:

/opt/conda/envs/rapids/lib/python3.7/site-packages/cudf/core/dataframe.py in _setitem_with_dataframe(input_df, replace_df, input_cols, mask)
   7266     if len(input_cols) != len(replace_df.columns):
   7267         raise ValueError(
-> 7268             "Number of Input Columns must be same replacement Dataframe"
   7269         )
   7270 

ValueError: Number of Input Columns must be same replacement Dataframe
```

**Steps/Code to reproduce bug**

I have prepared a [Drive folder](https://drive.google.com/drive/u/1/folders/1iUb-hdAT4x8GbpavcnitymK4dOEAmyWL) (internal-only for NVIDIA) with an example parquet file and notebook to reproduce the error. Here is a snippet of that code.

```
ds = nvt.Dataset(INPUT_PATH, engine="parquet", part_size="1000MB")

cat_names = []
cont_names = ['NetworkTestLatency']
label_name = []

# Initalize our Workflow
workflow = nvt.Workflow(cat_names=cat_names, cont_names=cont_names, label_name=label_name)


# Add Normalize to the workflow for continuous columns
workflow.add_cont_feature(nvt.ops.Normalize())

# Finalize the Workflow
workflow.finalize()

%%time
    
workflow.apply(
    ds,
    output_format="parquet",
    output_path=OUTPUT_PATH,
    shuffle=None,
    out_files_per_proc=1,
)
```

**Environment details (please complete the following information):**
NVT 0.3

**Comment by rnyak** [comment]
This looks like a workaround to solve this issue;

`df_inspect['NetworkTestLatency'] = df_inspect['NetworkTestLatency'].astype('int64')` 

The original `float32` values are somehow yield error.

**Comment by rnyak** [comment]
This is because var and std returns as `-inf` and `nan` respectively.  Therefore the condition below does not met so new_gdf returns  as an empty df without a column. :

```for name in cont_names:
            if stats_context["stds"][name] > 0:
                new_col = f"{name}_{self._id}"
                new_gdf[new_col] = (gdf[name] - stats_context["means"][name]) / (
                    stats_context["stds"][name]
                )
                new_gdf[new_col] = new_gdf[new_col].astype("float32")
```
And this creates issue/error "Number of Input Columns must be same replacement Dataframe" here:
``` def assemble_new_df(self, origin_gdf, new_gdf, target_columns):
        if self.replace and self.preprocessing and target_columns:
            if new_gdf.shape[0] < origin_gdf.shape[0]:
                return new_gdf
            else:
                origin_gdf[target_columns] = new_gdf

**Comment by benfred** [comment]
This is fixed by #444
